src/app/Containers/PreviewContainer/index.unit.test.js

```diff
@@ -0,0 +1,60 @@
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { Provider } from 'react-redux';
+import { mount } from 'enzyme';
+
+import { SUCCESS, ERROR, LOADING } from '../../../common/constants';
+import PreviewContainer from '.';
+import { Error, Spinner } from '../../../common/Components';
+
+const middlewares = [thunk];
+const mockStore = configureStore(middlewares);
+
+const getContainerWithSpecificState = (state) => {
+  const store = mockStore(state);
+  let container = mount(
+    <Provider store={store}>
+      <PreviewContainer />
+    </Provider>,
+  );
+
+  return container;
+};
+
+describe('PreviewContainer', () => {
```

Comment:
I think we should restructure this tests a little. As we have talked, the setting up of the tests should be on a describe, not on the `it`s. So, as in this case we have three separate scenarios, maybe we could do something like this:
```
describe('PreviewContainer', () => {
   // global setup

   describe('when the state value is LOADING', () => {
      // local setup
      it('Renders the Spinner component', () => { ... }
   })

   describe('when the state value is ERROR', () => {
      // local setup
      it('Renders the Error component', () => { ... ​}
   })

   describe('when the state value is SUCCESS', () => {
      // local setup
     it('Renders the PreviewContainer , () => { ... ​}  
   })
})
```

What do you think?